### PR TITLE
glTF 2.0 morph animation support

### PR DESCRIPTION
**Pull request by takahirox**
This PR lets `GLTF2Loader` support glTF 2.0 morph animation, excluding `TANGENT`. #11326

I'm not familiar with tangent yet.
How can we support tangent?
Converting tangent to normal?
Let's make another PR for `TANGENT` support.

### CODE REVIEW

File: examples/js/loaders/GLTF2Loader.js [hunk 1]
```diff
@@ -2111,6 +2112,92 @@ THREE.GLTF2Loader = ( function () {
...
+									//     + weight0 * ( morphTarget0 - position )
+									//     + weight1 * ( morphTarget1 - position )
+									//     ...
+									// while the glTF one is
+									//   position
+									//     + weight0 * morphTarget0
+									//     + weight1 * morphTarget1
+									//     ...
+									// then adding position to morphTarget.
+									// motphTarget should depend on mesh, so cloning attribute.
```
Comment (donmccurdy) [inline examples/js/loaders/GLTF2Loader.js]: *morphTarget

These BufferAttributes aren't mutated while playing animation, are they? If not, what's the reason for cloning them? I'm not sure I follow the "should depend on mesh" part.

Comment (takahirox) [inline examples/js/loaders/GLTF2Loader.js]: Thanks, I've updated the comment to clarify more.

**Comment by donmccurdy** [comment]
Nice! Does this work with both the sphere and cube examples?

**Comment by takahirox** [comment]
Yup! But we also need to merge #11351 for them.

**Comment by donmccurdy** [comment]
Ok cool! May also want to wait on this PR until #11285 is in.

File: examples/js/loaders/GLTF2Loader.js [hunk 2]
```diff
@@ -2111,6 +2112,93 @@ THREE.GLTF2Loader = ( function () {
 						meshNode = new THREE.Mesh( geometry, material );
 						meshNode.castShadow = true;
 
+						if ( primitive.targets !== undefined ) {
+
+							geometry.morphTargets = [];
```
Comment (Mugen87) [inline examples/js/loaders/GLTF2Loader.js]: With #11285 is merged, creating `geometry.morphTargets = [];` should not be necessary anymore. Instead, you can set the name of the morph target directly to a `morphAttribute`, see https://github.com/mrdoob/three.js/issues/11277#issuecomment-299420508.

Comment (mrdoob) [inline examples/js/loaders/GLTF2Loader.js]: #11285 has been merged 😊

Comment (Mugen87) [inline examples/js/loaders/GLTF2Loader.js]: Yay 🎉 ! 😊

**Comment by takahirox** [comment]
Updated!

**Comment by takahirox** [comment]
Updated for the case if some `target` in `targets` don't have all attributes, like this

    "targets": [
        {
            "POSITION": 0
        },
        {
            "NORMAL": 1
        }
    ]

**Comment by mrdoob** [comment]
> Updated for the case if some `target` in `targets` don't have all attributes

@donmccurdy looks good?

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**